addon/ajax-request.js

```diff
@@ -42,6 +42,35 @@ function isJSONAPIContentType(header) {
   return header.indexOf(JSONAPIContentType) === 0;
 }
 
+function startsWithSlash(string) {
+  if (typeof string === 'string') {
+    return string.charAt(0) === '/';
+  }
+  return string;
+}
+
+function endsWithSlash(string) {
+  if (typeof string === 'string') {
+    return string.charAt(string.length - 1) === '/';
+  }
+  return string;
+}
+
+function stripSlashes(path) {
+  if (typeof path === 'string') {
```

Comment:
Instead of checking for string 3 times, you could either type cast the argument to a string or throw an error if the argument it not a string.